### DISCUSSION TITLE
Axios Version unable to updating

### DISCUSSION THREAD

**Discussion by vijay9561** [axios/axios#6370] [General]
Hi Team,

I am updating axios **0.27.2** to **1.6.0** after that I am experiencing an below issue on code compilation:

Types of property 'headers' are incompatible
Index signatures are incompatible.
Type 'AxiosHeaderValue' is not assignable to type 'string'.
Type 'number' is not assignable to type 'string'

**Comment by jasonsaayman** [comment]
Axios v1 has stricter header types. Header values are not typed as only `string` anymore; they can also be `number`, `boolean`, `null`, arrays, or managed through `AxiosHeaders`.

This usually happens when app code has a custom type like:

```ts
headers: Record<string, string>
```

but Axios v1 expects Axios header types.

Prefer using Axios' exported types instead:

```ts
import type { RawAxiosRequestHeaders } from 'axios';

const headers: RawAxiosRequestHeaders = {
  Authorization: `Bearer ${token}`,
  'Content-Type': 'application/json'
};
```

or use `AxiosHeaders`:

```ts
import { AxiosHeaders } from 'axios';

const headers = new AxiosHeaders();
headers.set('Authorization', `Bearer ${token}`);
```

If this is in an interceptor, use `InternalAxiosRequestConfig` for the interceptor config and mutate headers through `set` when available:

```ts
axios.interceptors.request.use(config => {
  config.headers.set('Authorization', `Bearer ${token}`);
  return config;
});
```

So the fix is usually to replace custom `Record<string, string>` header annotations with Axios' header types.
